### PR TITLE
(PA-1503) Update curl to v7.56.1

### PR DESCRIPTION
**Pull request by caseywilliams**
7.56.1 came out just after we merged the bump to 7.56.0

### CODE REVIEW

File: configs/components/curl.rb [hunk 1]
```diff
@@ -1,5 +1,5 @@
 component 'curl' do |pkg, settings, platform|
-  pkg.version '7.56.0'
+  pkg.version '7.56.1'
   pkg.md5sum '65351b9df687ed539852ae7b9464006c'
```
Comment (MikaelSmith) [inline configs/components/curl.rb]: Need to update the checksum. I suggest testing builds on a few platforms: a linux, windows, maybe solaris/aix.

**Comment by ScottGarman** [comment]
Yep, this will need a link to a successful ad-hoc build on pretty much all of our platforms.

File: configs/components/curl.rb (hunk 1 above)
Comment (caseywilliams) [reply configs/components/curl.rb]: Gah, thanks @MikaelSmith, I'll update and push again once I've successfully built this on a few platforms.

**Comment by caseywilliams** [comment]
Okay, PR updated with the md5sum, and here's an ad-hoc build where curl succeeded for a selection of Linux, Solaris, AIX, Mac, and Windows: https://jenkins-master-prod-1.delivery.puppetlabs.net/view/puppet-agent/view/ad-hoc/job/platform_puppet-agent_puppet-agent-packaging_daily-ad_hoc/2/

**Pull request merged**